### ISSUE TITLE
NAN on this API combination

### ISSUE THREAD

**Issue by beanduan22** [pytorch/pytorch#114037]
### 🐛 Describe the bug
```python
import torch
import torchvision
import torchvision.transforms as transforms
import torch.nn as nn
import os
import torch.nn.functional as F
gpu_device = torch.device("cuda")
cpu_device = torch.device("cpu")
class PreprocessAndCalculateModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.batchnorm2d = nn.BatchNorm2d(3)
    def forward(self, x):

        x = self.batchnorm2d(x)
        x_ = torch.linalg.slogdet(x)
        x_erfinv = torch.special.erfinv(x)
        return x_erfinv

real_inputs = torch.rand(2, 3, 8, 8)
model = PreprocessAndCalculateModel()
x = real_inputs
output_gpu = model.to(gpu_device)(x.cuda())
output_cpu = model.to(cpu_device)(x.cpu())
print(torch.isnan(output_gpu).any())
print(torch.isnan(output_gpu).any())
```
### Outputs
```
tensor(True, device='cuda:0')
tensor(True, device='cuda:0')
```

**Comment by abastola0** [comment]
can you please assign me this issue?

**Comment by ZailiWang** [comment]
https://pytorch.org/docs/stable/special.html#torch.special.erfinv
During the test please make sure your input is valid.

**Comment by malfet** [comment]
`torch.erfiinv` is defined only in range [-1, 1], outside of that range it returns nan. And `batchnorm2d` was never guaranteed to return values in that range:
```
$ python3 -c "import torch;print(torch.special.erfinv(torch.tensor(3.14)))"
tensor(nan)
```
